Staging for booking flow on the client side dashboard page

byte-man74 opened the pull request:
- completed the flow for user booking to 70%
- the last flow here is the paystack setup, the viewing of reciepts for a specific booking and also the cancellation of user booking.
-

byte-man74 committed:
refactor: remove unused product view page and update package view component

- Deleted the obsolete product view page from the client dashboard.
- Renamed the imported ProductForm to package-form in the package view component.
- Enhanced the Package component to include session handling for routing based on user authentication.
- Updated routing logic to direct users to the appropriate package details based on their session status.
- Cleaned up commented-out code in the Package component for better readability.

byte-man74 committed:
feat: :sparkles: Enhance package management and booking features

- Added `react-intersection-observer` for improved lazy loading.
- Introduced new API routes for viewing user bookings.
- Updated package services to support pagination with page parameters.
- Enhanced the package view component to include a "Book Now" button for users.
- Improved the skeleton loading UI for package details.
- Updated client dashboard navigation to include a new "Bookings" section.
- Refactored package listing to support infinite scrolling and dynamic loading of packages.

byte-man74 committed:
feat: :sparkles: Implement booking initiation and enhance booking table

- Added new API route for initiating bookings with dynamic package ID.
- Introduced `InitiateBookingRequest` type for booking requests.
- Enhanced `BookingTable` component with improved UI and status handling.
- Updated `InitiateBookingForm` to handle booking initiation and display success/error messages.
- Refactored loading skeletons and added icons for better visual representation.
- Cleaned up sidebar component by commenting out unused Breadcrumbs.

byte-man74 committed:
Merge branch 'develop' into staging-for-booking

byte-man74 committed:
feat: :sparkles: Enhance booking features and UI improvements

- Added new API route for viewing and editing bookings with dynamic booking ID.
- Introduced `BookingInformationResponse` type for detailed booking information.
- Updated `BookingTable` component to include a link for viewing booking details.
- Improved sidebar and user navigation components for better user experience.
- Refactored styles in the NotFound page for consistency.
- Cleaned up unused imports and ensured proper handling of user session data.

byte-man74 committed:
feat: :sparkles: Add booking payment functionality and enhance booking details view

- Introduced new API routes for initiating and verifying booking payments.
- Added `BookingVerifyPaymentResponse` and `InitiateBookingPaymentRequest` types for payment handling.
- Updated `BookingView` component to include payment progress and details.
- Enhanced UI elements for better user experience in the booking details section.
- Improved routing to support payment initiation from the booking view.

byte-man74 committed:
feat: :sparkles: Add Booking Transactions page

- Introduced a new BookingTransactionsPage component for displaying booking transaction details.
- Implemented basic layout with a header and a placeholder for transaction details.
- Enhanced the client dashboard by integrating the new transactions page for improved user navigation.

byte-man74 committed:
feat: :sparkles: Update booking payment verification and enhance booking details

- Refactored the `verifyBookingPayment` route to use a dynamic reference parameter.
- Updated `BookingInformationResponse` type to include detailed transaction information.
- Enhanced the `BookingPayment` component to handle payment references as strings.
- Added a modal in the `BookingView` component to display payment receipts with improved UI.
- Improved state management for booking references and payment plans in the booking components.

profstrategy commented on api/services/booking.ts:
```diff
@@ -0,0 +1,48 @@
+
+
+export const useInitiateBookingPayment = (bookingId: string, body?: InitiateBookingPaymentRequest) => {
+    return useAppMutation<any>({
```
Why passing any here, the type here should mimic shape of request type

byte-man74 replied:
Thanks I'll address this. Using "any" type here is not good practice

byte-man74 committed:
feat: :sparkles: Add cancel booking functionality and enhance booking payment process

- Introduced a new API route for canceling bookings with dynamic booking ID.
- Updated `BookingView` component to include a confirmation modal for booking cancellations.
- Enhanced `BookingPayment` component to support payment method selection (card or wallet).
- Updated types to include `InitiateBookingApiResponse` for better handling of booking initiation responses.
- Improved UI elements for payment information and added wallet balance checks during payment.
- Refactored `InitiateBookingForm` to invalidate queries upon successful booking initiation.

byte-man74 committed:
feat: :sparkles: Enhance booking management and UI components

- Added `is_active` field to `BookingInformationResponse` for better booking status handling.
- Updated `BookingView` component to display a modal when a booking is deactivated, improving user feedback.
- Refactored `BookingTable` component to use dynamic routing for package links, enhancing navigation.
- Cleaned up imports in `BookingTable` for better code organization.

byte-man74 committed:
feat: :sparkles: Add wallet payment option and enhance booking payment process

- Introduced a new API route for making bookings through wallet payments.
- Updated `BookingPayment` component to support wallet payment method alongside card payments.
- Enhanced payment handling with improved state management and query invalidation for wallet transactions.
- Refactored `BookingView` component to use `Link` for downloading receipts, improving navigation.
- Updated `useCreatePackage` to ensure data is stringified before sending in requests.
